include/Dialect/Poly/IR/PolyOps.td

```diff
@@ -44,16 +44,16 @@ def Poly_MulOp : Poly_BinOp<"mul"> {
   let summary = "Multiplication operation between polynomials.";
 }
 
-def Poly_MulConstantOp : Poly_Op<"mul_constant"> {
+def Poly_MulConstantOp : Poly_Op<"mul_constant", [ElementwiseMappable]> {
   let summary = "Multiplication by a constant of the field.";
 
   let arguments = (ins
-    Polynomial:$x,
-    AnyInteger:$constant
+    PolynomialLike:$x,
+    AnyIntegerLike:$constant
```

Comment:
We may want both of these, but this one I'd also have expected a single constant to be multiplied across a `PolynomialLike` - e.g. Take `(a, b)` as a TensorOf<Poly> and multiply by the constant `q`, to retrieve `(q*a, q*b)`.

Reply:
Simpler is better, done.